components/catalogue/EditableProduct.jsx

```diff
@@ -0,0 +1,50 @@
+import { useState } from "react";
+import EditProduct from "./EditProduct.jsx";
+import ProductDetails from "./ProductDetails.jsx";
+
+function EditableProduct(props) {
+  const [isEditing, setIsEditing] = useState(false);
+  const [product, setProduct] = useState(props.product);
+
+  function saveProduct(newProduct) {
+    setProduct((prevProduct) => {
+      let res = Object.assign({}, prevProduct);
+      res.title = newProduct.title;
+      res.price = newProduct.price;
+      res.description = newProduct.description;
+      res.categories = newProduct.categories;
+      res.picture = newProduct.picture;
+      res.updatedAt = Date();
+
+      return res;
+    });
+
+    // TODO: Check that the user can edit the product used as parameter
+    // TODO: If "validation" is true
+    setIsEditing(false);
+  }
```

Comment:
I think you can do something like this:
 ```suggestion
    setProduct((prevProduct) => ({
      ...prevProduct, 
      title: newProduct.title, 
      price: newProduct.price,
      // rest of props
    }));
    setIsEditing(false);
  }
```

Basically what we are doing here is taking the previous object, [spreading](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) its properties on a new object, and replacing those with new ones